pkg/policy/rule.go

```diff
@@ -269,6 +269,7 @@ func (r *rule) resolveL4IngressPolicy(ctx *SearchContext, state *traceState, res
 				ruleCopy.FromEndpoints[idx].MatchExpressions = append(ruleCopy.FromEndpoints[idx].MatchExpressions, requirements...)
 				ruleCopy.FromEndpoints[idx].SyncRequirementsWithLabelSelector()
 			}
+			ruleCopy.SetAggregatedSelectors()
```

Comment:
I'm not a fan of this - I think we need to rework how we deal with requirements. The existing implementation is already not very performant :(

Reply:
Agreed but this is not the end of the world for now